### PR TITLE
key chains: Keep listeners when decrypting/encrypting

### PR DESCRIPTION
**Pull request by oscarguindzberg**

### CODE REVIEW

File: core/src/main/java/org/bitcoinj/wallet/BasicKeyChain.java [hunk 1]
```diff
@@ -475,6 +479,9 @@ public BasicKeyChain toEncrypted(KeyCrypter keyCrypter, KeyParameter aesKey) {
                     throw new KeyCrypterException("The key " + key.toString() + " cannot be successfully decrypted after encryption so aborting wallet encryption.");
                 encrypted.importKeyLocked(encryptedKey);
             }
+            for (ListenerRegistration<KeyChainEventListener> listener : listeners) {
+                encrypted.listeners.add(listener);
```
Comment (schildbach) [inline core/src/main/java/org/bitcoinj/wallet/BasicKeyChain.java:483]: You could use the newly added `addEventListener(ListenerRegistration)` here, no?

File: core/src/main/java/org/bitcoinj/wallet/BasicKeyChain.java [hunk 2]
```diff
@@ -500,6 +507,9 @@ public BasicKeyChain toDecrypted(KeyParameter aesKey) {
             for (ECKey key : hashToKeys.values()) {
                 decrypted.importKeyLocked(key.decrypt(aesKey));
             }
+            for (ListenerRegistration<KeyChainEventListener> listener : listeners) {
```
Comment (schildbach) [inline core/src/main/java/org/bitcoinj/wallet/BasicKeyChain.java:510]: …and here?

**Review by schildbach — COMMENTED** [review]
Looks like a good catch!

File: core/src/main/java/org/bitcoinj/wallet/BasicKeyChain.java (hunk 1 above)
Comment (oscarguindzberg) [reply core/src/main/java/org/bitcoinj/wallet/BasicKeyChain.java]: done.

File: core/src/main/java/org/bitcoinj/wallet/BasicKeyChain.java (hunk 2 above)
Comment (oscarguindzberg) [reply core/src/main/java/org/bitcoinj/wallet/BasicKeyChain.java:510]: done.

**Comment by schildbach** [comment]
I changed the visibility of the method to package-private and merged.

**Pull request closed without merging**